lib/SocketHelpers.cpp

```diff
@@ -147,9 +162,23 @@ std::optional<Message> ReceiveProxyWifiMessage(const wil::unique_socket& socket)
 
 static void SendBytes(wil::unique_socket& socket, gsl::span<const uint8_t> dataToSend)
 {
-    // TODO: Is it needed to loop on a send?
     while (dataToSend.size_bytes() > 0)
     {
+        // Wait until the destination is ready to receive
+        const timeval timeout{0, 500000}; // 0.5 sec timeout
```

Comment:
Can this be defined outside the loop?

Reply:
I'd rather keep it in the smallest possible context. The compiler will optimize that anyway. I can make it a constexpr though.